packages/core/src/context/experiment/experiment-reducers.ts

```diff
@@ -227,14 +264,35 @@ export const experimentReducer = produce(
       case 'editCategoricalVariable':
         state.categoricalVariables[action.payload.index] =
           experimentSchema.shape.categoricalVariables.element.parse(
-            action.payload.variable
+            action.payload.newVariable
           )
+        state.dataPoints = updateDataPointNames(
+          state,
+          action.payload.oldName,
```

Comment:
Why not read the old name from the current state instead of requiring the client to know about how the reducer works?

Reply:
Good question. I'll change it to find the name using the index that is in the payload anyway.